packages/api/src/external/carequality/patient-demographics.ts

```diff
@@ -0,0 +1,89 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { LinkDemographics } from "@metriport/core/domain/patient-demographics";
+import { PatientResource } from "@metriport/ihe-gateway-sdk";
+import {
+  scoreLink,
+  linkHasNewDemographics,
+  patientToNormalizedCoreDemographics,
+  removeInvalidArrayValues,
+  normalizeDob,
+  normalizeGender,
+  normalizeAndStringifyNames,
+  normalizeAddress,
+  stringifyAddress,
+  normalizeTelephone,
+  normalizeEmail,
+} from "../../domain/medical/patient-demographics";
+import { CQLink } from "./cq-patient-data";
+
+export function getNewDemographics(patient: Patient, links: CQLink[]): LinkDemographics[] {
+  const coreDemographics = patientToNormalizedCoreDemographics(patient);
+  const consolidatedLinkDemographics = patient.data.consolidatedLinkDemographics;
+  return getPatientResources(links)
+    .map(patientResourceToNormalizedLinkDemographics)
+    .filter(linkDemographics => scoreLink({ coreDemographics, linkDemographics })[0])
+    .filter(linkDemographics =>
+      linkHasNewDemographics({
+        coreDemographics,
+        consolidatedLinkDemographics,
+        linkDemographics,
+      })
+    );
+}
+
+export function patientResourceToNormalizedLinkDemographics(
+  patientResource: PatientResource
+): LinkDemographics {
+  const dob = normalizeDob(patientResource.birthDate);
+  const gender = normalizeGender(patientResource.gender);
+  const names = (patientResource.name ?? []).flatMap(name => {
+    return (name.given ?? []).map(firstName => {
+      return normalizeAndStringifyNames({ firstName, lastName: name.family });
+    });
+  });
+  const addresses = (patientResource.address ?? []).map(a => {
+    return stringifyAddress(
+      normalizeAddress({
+        line: a.line,
+        city: a.city,
+        state: a.state,
+        zip: a.postalCode,
+        country: a.country,
+      })
+    );
+  });
+  const telephoneNumbers = (patientResource.telecom ?? []).flatMap(tc => {
+    if (!tc.value || !tc.system) return [];
+    if (tc.system !== "phone") return [];
+    return [normalizeTelephone(tc.value)];
+  });
+  const emails = (patientResource.telecom ?? []).flatMap(tc => {
+    if (!tc.value || !tc.system) return [];
+    if (tc.system !== "email") return [];
+    return [normalizeEmail(tc.value)];
```

Comment:
So from looking at the data in the pd results table, its very common that both phones and emails dont have their system parameter defined. We can try to resolve that in the ihe v2 code, or we can resolve that here and have some sort of logic to determine if something is a phone or email i.e. emails have @ signs in them or whatever

Reply:
interesting -- I was going to ask whether that was deployed! 

I think ultimately here is likely the best place. Will try to address today.